### PR TITLE
HDDS-10218. Speed up TestSstFilteringService

### PR DESCRIPTION
**Pull request by raju-balpande**
## What changes were proposed in this pull request?
Changes are done to improve the test run speed to 33.23 s as compare to 59.43 s earlier.
- The setup method modified to be initialize one time for all methods.
- Introduced the method orderer
- Declared one counter to track the number of snapshots been generated in all the methods to assert the count according.
- Separated the methods for creation of volume and bucket which was combine and there were repeated calls to it.
- Changed one assert statement to assert that because the .sst file from the bucket created in first method was not consistent to available in second method.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10218

## How was this patch tested?
Tested on local system and in CI-CD integration.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java [hunk 1]
```diff
@@ -180,13 +183,14 @@ public void testIrrelevantSstFileDeletion()
...
     String bucketName2 = "buck2";
-    createVolumeAndBucket(volumeName, bucketName2);
+    addBucketToVolume(volumeName, bucketName2);
     createKeys(volumeName, bucketName2, keyCount);
 
     activeDbStore.getDb().flush(OmMetadataManagerImpl.KEY_TABLE);
     List<LiveFileMetaData> allFiles = activeDbStore.getDb().getSstFileList();
     String snapshotName1 = "snapshot1";
     writeClient.createSnapshot(volumeName, bucketName2, snapshotName1);
+    countTotalSnapshots++;
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java:193]: I suggest extracting these two statements into a method (and using it for other similar two-liners) to:
- ensure count is incremented for each snapshot
- reduce duplication

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java [hunk 2]
```diff
@@ -71,24 +75,21 @@
 /**
  * Test SST Filtering Service.
  */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(OrderAnnotation.class)
 public class TestSstFilteringService {
-  public static final String SST_FILE_EXTENSION = ".sst";
-  @TempDir
-  private File folder;
+  private static final String SST_FILE_EXTENSION = ".sst";;
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java:81]: nit:

```suggestion
  private static final String SST_FILE_EXTENSION = ".sst";
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @raju-balpande for the patch.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java (hunk 1 above)
Comment (raju-balpande) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java]: Done with the changes.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java (hunk 2 above)
Comment (raju-balpande) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestSstFilteringService.java]: Done. thanks.

**Review by adoroszlai — COMMENTED** [review]
Thanks @raju-balpande for updating the patch, LGTM.

**Comment by hemantk-12** [comment]
> I would avoid tests ordering. 

+1 if it is possible.

Otherwise patch looks good to me.

**Comment by raju-balpande** [comment]
The methods are independent of ordering flow when snapshots are concerned. Independent method run also works fine. but the ordering was required because checks on key operations are dependent. Clearing all the data again will consume time and there will be hardly any improvement in time. but I am trying to see if I can fix it for keys.

**Comment by adoroszlai** [comment]
Thanks @raju-balpande for the patch, @hemantk-12, @myskov for the review.

Let's work on improving independence of test methods separately.

**Pull request merged**